Description: Fix segfault when renewing station certificate
 Thank you to Rick Murphy K1MU/4 for the patch
Comment: This patch can be removed when tqsl 2.6.4 is released
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1012020
Forwarded: not-needed

--- a/apps/ChangeLog.txt
+++ b/apps/ChangeLog.txt
@@ -1,5 +1,11 @@
 TQSL changes
 ============
+2.6.4
+----------
+Correct a crash when TQSL detects a callsign certificate that's about to
+expire during startup and the user selects "Yes" to renew that callsign
+certificate.
+
 2.6.3
 ----------
 Handle the case where there's both a valid and an invalid callsign certificate
--- a/apps/certtree.cpp
+++ b/apps/certtree.cpp
@@ -202,6 +202,7 @@
 	// Iterate the tree, looking for a matching certificate
 	wxTreeItemId root = GetRootItem();
 	wxTreeItemIdValue issCookie;
+	wxTreeItemIdValue listsCookie;
 	wxTreeItemIdValue certCookie;
 	wxTreeItemId top;
 	if (_nissuers > 1) {
@@ -210,16 +211,20 @@
 		top = root;
 	}
 	while (top.IsOk()) {
-		wxTreeItemId item = GetFirstChild(top, certCookie);
+		wxTreeItemId item = GetFirstChild(top, listsCookie);
 		while (item.IsOk()) {
-			tQSL_Cert cert = GetItemData(item)->getCert();
-			long s;
-			tqsl_getCertificateSerial(cert, &s);
-			if (s == serial) {	// found it
-				SelectItem(item);
-				return;
+			wxTreeItemId subitem = GetFirstChild(item, certCookie);
+			while (subitem.IsOk()) {
+				tQSL_Cert cert = GetItemData(subitem)->getCert();
+				long s;
+				tqsl_getCertificateSerial(cert, &s);
+				if (s == serial) {	// found it
+					SelectItem(subitem);
+					return;
+				}
+				subitem = GetNextChild(item, certCookie);
 			}
-			item = GetNextChild(top, certCookie);
+			item = GetNextChild(top, listsCookie);
 		}
 		if (_nissuers > 1) {
 			top = GetNextChild(root, issCookie);
